External/FEXCore/unittests/Emitter/ALU_Tests.cpp

```diff
@@ -782,23 +782,26 @@ TEST_CASE_METHOD(TestDisassembler, "Emitter: ALU: AddSub - shifted register") {
     TEST_SINGLE(add(Size::i64Bit, Reg::r30, Reg::r29, Reg::r28, ShiftType::LSL, 63), "add x30, x29, x28, lsl #63");
     TEST_SINGLE(add(Size::i32Bit, Reg::r30, Reg::r29, Reg::r28, ShiftType::LSL, 31), "add w30, w29, w28, lsl #31");
 
-    TEST_SINGLE(add(Size::i32Bit, Reg::r30, Reg::r29, Reg::r28, ShiftType::LSL, 32), "unallocated (Unallocated)");
+    // Vixl decodes this as a "valid" instruction, should be "unallocated (Unallocated)".
+    TEST_SINGLE(add(Size::i32Bit, Reg::r30, Reg::r29, Reg::r28, ShiftType::LSL, 32), "add w30, w29, w28, lsl #32");
```

Comment:
I'd consider removing these instead... I agree it's a Vixl bug, but I'm dubious of the premise of the original test ("if we put rubbish in, we want specific rubbish out"). If it's going to generate a bad instruction, I'd add an assert in the generating function instead?

Reply:
You're right. I added an assert to the instruction emitters and removed the tests.